project1_tiny_calculator/lib/calculate_provider.dart

```diff
@@ -0,0 +1,47 @@
+import 'package:flutter/material.dart';
+
+class CalculateProvider extends ChangeNotifier {
+  List<String> operations = ['+', '/', '-', '*'];
+  final TextEditingController firstNumberController = TextEditingController();
```

Comment:
should we have text editing controllers in providers?

Reply:
yes it wouldn't be that problematic.